src/attributes.md

```diff
@@ -82,6 +81,47 @@ fn some_unused_variables() {
 }
 ```
 
+There are three kinds of attributes:
+
+* Built-in attributes
+* Macro attributes
+* Derive mode helper attributes
+
+## Dynamic and inert attributes
+
+An attribute is either dynamic or inert. During attribute processing, *dynamic
+attributes* remove themselves from the thing they are on while *inert attriutes*
+stay on.
+
+The `cfg` and `cfg_attr` attributes are dynamic. The `test` attribute is inert
```

Comment:
The full list is `cfg`, `cfg_attr`, `test`, `bench`, `derive` currently, if I'm not missing anything.

`test` and `bench` are being migrated to macros right now though (https://github.com/rust-lang/rust/pull/53410).
(Also, `test`/`bench` are always "dynamic", they tweak their target items in test mode as well - add `pub`s, reexports.)